src/commons/application/Application.tsx

```diff
@@ -237,6 +240,49 @@ const Application: React.FC<ApplicationProps> = props => {
           </Switch>
         )}
       </div>
+
+      {/* agreedToResearch has a default value of undefined in the store.
+          It will take on null/true/false when the backend returns. */}
+      {props.agreedToResearch === null && (
+        <div className="research-prompt">
+          <Dialog
```

Comment:
Im not sure if this is the best way to handle this. This PR (and the backend https://github.com/source-academy/backend/pull/790) does not take into account instructors who run their own Source Academy course on their own backend. In such instances, the prompt should not display (and ideally the backend should not track or send this boolean either)

Reply:
another env variable in the frontend?

Reply:
@chownces Yeah, should add a flag for this.